pixeltable/share/packager.py

```diff
@@ -343,22 +343,35 @@ def __import_table(self, bundle_path: Path, tv: catalog.TableVersion, tbl_md: sc
         # previously imported replica.
         # The pxtmedia:// URLs of local media files are constructed using the SHA256 hash of the media file contents,
         # so this works for media files too, providing bytewise validation of data integrity.
-        non_vmax_temp_cols = [col for col_name, col in temp_cols.items() if col_name != 'v_max']
-        non_vmax_store_cols = [store_sa_tbl.c[col_name] for col_name in temp_cols if col_name != 'v_max']
-        q = sql.select(*non_vmax_temp_cols, *non_vmax_store_cols).where(pk_clause)
+
+        pk_col_names = {col.name for col in tv.store_tbl.pk_columns()}
+        value_store_cols = [
+            store_sa_tbl.c[col_name] for col_name in temp_cols if col_name not in pk_col_names and col_name != 'v_max'
+        ]
+        value_temp_cols = [
+            col for col_name, col in temp_cols.items() if col_name not in pk_col_names and col_name != 'v_max'
+        ]
+        mismatch_predicates = [store_col != temp_col for store_col, temp_col in zip(value_store_cols, value_temp_cols)]
+        mismatch_clause = sql.or_(*mismatch_predicates)
+
+        # This query looks for rows that have matching primary keys (rowid + v_min), but differ in at least one value
+        # column.
+        q = sql.select(*value_store_cols, *value_temp_cols).where(pk_clause).where(mismatch_clause)
```

Comment:
Would be good to add the SQL as an explanatory comment, along the lines of what's in catalog.py:load_tbl_md().

Reply:
Done.